MobileBuy/buy/src/main/java/com/shopify/buy/utils/AndroidPayHelper.java

```diff
@@ -0,0 +1,391 @@
+        // The merchant name will be shown on the top of the Android Pay dialogs
+        return MaskedWalletRequest.newBuilder()
+                .setMerchantName(merchantName)
+                .setPhoneNumberRequired(phoneNumberRequired)
+                .setShippingAddressRequired(checkout.isRequiresShipping())
+                .setCurrencyCode(checkout.getCurrency())
+                .setEstimatedTotalPrice(checkout.getTotalPrice())
+                .setPaymentMethodTokenizationParameters(parameters)
+                .setCart(walletCart)
+                .setPhoneNumberRequired(phoneNumberRequired)
+                .build();
+    }
+
+    /**
+     * Creates a Full Wallet Request from a Shopify Checkout
+     *
+     * @param checkout     The {@link Checkout} to use, not null.
+     * @param maskedWallet A {@link MaskedWallet} returned from a previous {@link MaskedWalletRequest} sent to Android Pay
+     * @return A {@link FullWalletRequest} for use in retrieving a {@link com.google.android.gms.wallet.FullWallet}
+     */
+    public static FullWalletRequest createFullWalletRequest(Checkout checkout, MaskedWallet maskedWallet) {
+        if (checkout == null) {
+            throw new NullPointerException("checkout cannot be null");
+        }
+
+        if (maskedWallet == null) {
+            throw new NullPointerException("maskedWallet cannot be null");
+        }
+
+        return FullWalletRequest.newBuilder()
+                .setGoogleTransactionId(maskedWallet.getGoogleTransactionId())
+                .setCart(createWalletCart(checkout))
+                .build();
+    }
+
+    /**
+     * Will update the local checkout with the Addresses and email from a Masked Wallet.
+     * If any of the City, Country Code, Province Code, or ZIP code have changed any shipping rate previously set
+     * on the Checkout will be invalid and set to null. The checkout should be updated on Shopify using {@link BuyClient#updateCheckout(Checkout, Callback)},
+     * and the shipping rate will need to be fetched again using {@link BuyClient#getShippingRates(String, Callback)}
+     *
+     * @param checkout     The {@link Checkout} to update
+     * @param maskedWallet The {@link MaskedWallet} to use the address from
+     */
+    public static void updateCheckoutAddressAndEmail(Checkout checkout, MaskedWallet maskedWallet) {
+        if (checkout == null) {
+            throw new NullPointerException("checkout cannot be null");
+        }
+
+        if (maskedWallet == null) {
+            throw new NullPointerException("maskedWallet cannot be null");
+        }
+
+        Address shippingAddress = createShopifyAddress(maskedWallet.getBuyerShippingAddress());
+        Address billingAddress = createShopifyAddress(maskedWallet.getBuyerBillingAddress());
+
+        // If the location has changed, we need to invalidate the shipping address
+        if (!shippingAddress.locationsAreEqual(checkout.getShippingAddress())) {
```

Comment:
Here I got NPE, I was trying to update checkout that didn't have shipping address yet